src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -1988,6 +1990,115 @@ public void storeMainTableNameFormatPreferences(MainTableNameFormatPreferences p
+        }
+
+        searchDialogTableColunns = columns;
+    }
+
+    private List<MainTableColumnModel> createSearchDialogColumns() {
+        if (this.searchDialogTableColunns == null) {
+            updateSearchDialogTableColumns();
+        }
+        return this.searchDialogTableColunns;
+    }
+
+    /**
+     * Creates the ColumnSortOrder from cache
+     *
+     * @return List containing only the the columns in its proper sort order
+     */
+    private List<MainTableColumnModel> createSearchDialogColumnSortOrder() {
+        if (this.searchDialogColumnSortOrder == null) {
+            updateSearchDialogColumnSortOrder();
+        }
+        return this.searchDialogColumnSortOrder;
+    }
+
+    /**
+     * Reloads the MainTableColumnSortOrder from scratch to cache
+     */
+    private void updateSearchDialogColumnSortOrder() {
+        List<MainTableColumnModel> columnsOrdered = new ArrayList<>();
+        getStringList(SEARCH_DIALOG_COLUMN_SORT_ORDER).forEach(columnName ->
+                searchDialogTableColunns.stream().filter(column ->
+                        column.getName().equals(columnName))
+                                .findFirst()
+                                .ifPresent(columnsOrdered::add));
+
+        searchDialogColumnSortOrder = columnsOrdered;
+    }
+
+    @Override
+    public SearchDialogColumnPreferences getSearchDialogColumnPreferences() {
+        return new SearchDialogColumnPreferences(
+                createSearchDialogColumns(),
+                createSearchDialogColumnSortOrder());
+    }
+
+    /**
+     * Stores the {@link SearchDialogColumnPreferences} in the preferences
+     *
+     * @param columnPreferences the preferences to store
+     */
+    public void storeSearchDialogColumnPreferences(SearchDialogColumnPreferences columnPreferences) {
+        putStringList(SEARCH_DIALOG_COLUMN_NAMES, columnPreferences.getColumns().stream()
+                                                     .map(MainTableColumnModel::getName)
+                                                     .collect(Collectors.toList()));
+
+        List<String> columnWidthsInOrder = new ArrayList<>();
+        columnPreferences.getColumns().forEach(column -> columnWidthsInOrder.add(column.widthProperty().getValue().toString()));
+        putStringList(SEARCH_DIALOG_COLUMN_WIDTHS, columnWidthsInOrder);
+
+        List<String> columnSortTypesInOrder = new ArrayList<>();
+        columnPreferences.getColumns().forEach(column -> columnSortTypesInOrder.add(column.sortTypeProperty().getValue().toString()));
+        putStringList(SEARCH_DIALOG_COLUMN_SORT_TYPES, columnSortTypesInOrder);
+
+        putStringList(SEARCH_DIALOG_COLUMN_SORT_ORDER, columnPreferences
+                .getColumnSortOrder().stream()
+                .map(MainTableColumnModel::getName)
+                .collect(Collectors.toList()));
+
+        // Update cache
+        searchDialogTableColunns = columnPreferences.getColumns();
+    }
```

Comment:
Why so complicated? Just use the existing methods for getMainTableColumnPreferences, give them a more generic name and make them private, then call them from getMainTableColumnPreferences and getSearchDialogColumnPreferences, but deliver the constants SEARCH_DIALOG_COLUMN_NAMES / MAIN_TABLE ... variables as arguments.

Another thing: if we make SEARCH_DIALOG_COLUMN_NAMES distinct from MAIN_TABLE_COLUMN_NAMES, we need to make them customizable, since otherwise they would always keep their default value.